src/Files.App.Storage/NativeStorage/NativeFolderWatcher.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Files.Sdk.Storage.LocatableStorage;
+using Files.Sdk.Storage.MutableStorage;
+using System.Collections.Specialized;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Files.App.Storage.NativeStorage
+{
+	/// <inheritdoc cref="IFolderWatcher"/>
+    public sealed class NativeFolderWatcher : IFolderWatcher
+	{
+		private FileSystemWatcher? _fileSystemWatcher;
+		private NotifyCollectionChangedEventHandler? _collectionChanged;
```

Comment:
Are we marking the `private` fields with a `_` ? While it's written in the coding rules, I haven't seen it being very enforced. Outside of this PR, but we'd really need to settle if we do or not.

Reply:
I think it's good time to revise on codestyle guidelines we have on the website. It's been quite some time since they have been last updated